diff --git a/perl/Makefile.PL b/perl/Makefile.PL
index 69be46f..711dfa8 100644
--- a/perl/Makefile.PL
+++ b/perl/Makefile.PL
@@ -80,11 +80,9 @@ Can be either environment variables or passed on command line like:
 
 EOF
 
-$SIG{__DIE__} = sub {
-    print STDERR "\n  ** ERROR: ", @_;
-    print STDERR "  Run perl Makefile.PL SWISHHELP for options\n\n";
-    exit 1;
-};
+END {
+    print STDERR "Run perl Makefile.PL SWISHHELP for options\n\n" if $?;
+}
 
 #----------------------------------------------------------------------------------
 
